File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/seek/aware/ProductClient.kt
package io.micronaut.kafka.docs.seek.aware

import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.Topic
import io.micronaut.context.annotation.Requires
import io.micronaut.kafka.docs.Product

@Requires(property = "spec.name", value = "ConsumerSeekAwareTest")
@KafkaClient
interface ProductClient {
    @Topic("wonderful-products")
    fun produce(product: Product)
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/seek/aware/ProductClient.py
from abc import ABC, abstractmethod

from micronaut.configuration.kafka.annotation import KafkaClient, Topic
from micronaut.context.annotation import Requires
from micronaut.kafka.docs.Product import Product


@Requires(property="spec.name", value="ConsumerSeekAwareTest")
@KafkaClient
class ProductClient(ABC):

    @Topic("wonderful-products")
    @abstractmethod
    def produce(self, product: Product) -> None:
        ...
